onnxruntime/core/util/qmath.h

```diff
@@ -51,9 +51,7 @@ void QGemm(
     concurrency::ThreadPool* thread_pool);
 
 inline float RoundHalfToEven(float input) {
-  std::fesetround(FE_TONEAREST);
-  auto result = std::nearbyintf(input);
-  return result;
+  return input - std::remainderf(input, 1.f);
```

Comment:
From cppreference:

`The IEEE floating-point remainder of the division operation x/y calculated by this function is exactly the value x - n*y, where the value n is the integral value nearest the exact value x/y. When |n-x/y| = ½, the value n is chosen to be even.`

Reply:
What's the performance impact? Is it slower?

Reply:
Following program:

```cpp
#include <cfenv>
#include <cmath>

inline float RoundHalfToEven(float input) {
    if (!std::isfinite(input)) {
        return input;
    }
    return input - std::remainderf(input, 1.f);
}

inline float RoundHalfToEven2(float input) {
    std::fesetround(FE_TONEAREST);
    auto result = std::nearbyintf(input);
    return result;
}

int main() {
    float result{};
    for (float x = 1; x < 3e+038; x *= 1.0000001) {
        result += RoundHalfToEven2(x);  // or RoundHalfToEven
    }
    return static_cast<int>(result);
}
```

Gives me:

|       | Previous | Proposed |
|-------|----------|----------|
| MSVC  | 41.928   | 18.527   |
| clang | 40.267   | 17.455   |

So new method is at least 2.3x faster on both compilers on my CPU.